lib/config.js

```diff
@@ -435,15 +435,13 @@ function normalizeSanitizeFieldNames (opts) {
   }
 }
 
-function normalizeCloudProvider (opts) {
-  const allowedValues = ['auto', 'gcp', 'azure', 'aws', 'none']
-  if (allowedValues.indexOf(opts.cloudProvider) === -1) {
-    opts.cloudProvider = 'auto'
-    if (opts.logger) {
-      opts.logger.warn(
-        'Unexpected cloudProvider value %s, changing to auto',
-        opts.cloudProvider
-      )
+function normalizeCloudProvider (opts, logger) {
+  if ('cloudProvider' in opts) {
+    const allowedValues = ['auto', 'gcp', 'azure', 'aws', 'none']
+    if (allowedValues.indexOf(opts.cloudProvider) === -1) {
+      logger.warn('Invalid "cloudProvider" config value %s, falling back to default %s',
+        opts.cloudProvider, DEFAULTS.cloudProvider)
+      opts.cloudProvider = DEFAULTS.cloudProvider
```

Comment:
I see here we've changed from using a hard coded `auto` to using the value set in `DEFAULTS` -- 👍 future proofing.